storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/retriever/AbstractJdbcQueryTemplater.java

```diff
@@ -0,0 +1,438 @@
+        filePath = new File(fileString);
+      } catch (URISyntaxException e) {
+        throw new RuntimeException(String.format("Could not parse file string %s", fileString), e);
+      }
+
+      Statement statement;
+      String tempTableForLoad = createTempTableName();
+      List<String> loadEntitiesQueries =
+          this.createLoadEntityQuery(entityTable, tempTableForLoad, filePath);
+
+      try {
+        statement = this.connection.createStatement();
+        for (String query : loadEntitiesQueries) {
+          statement.execute(query);
+        }
+      } catch (SQLException e) {
+        throw new RuntimeException(
+            String.format(
+                "Could not load entity data from %s into table %s using query: \n%s",
+                filePath, entityTable, loadEntitiesQueries),
+            e);
+      }
+    }
+  }
+  /**
+   * Load entity rows from filePath to the destinationTable
+   *
+   * @param destinationTable
+   * @param temporaryTable temporary table for staging
+   * @param filePath csv file contains entity rows, with columns: entity_id and created_timestamp
+   * @return
+   */
+  protected List<String> createLoadEntityQuery(
+      String destinationTable, String temporaryTable, File filePath) {
+    List<String> queries = new ArrayList<>();
+    queries.add(
+        String.format("CREATE TABLE %s AS (SELECT * FROM %s);", temporaryTable, destinationTable));
+    //      queries.add(String.format("ALTER TABLE %s DROP COLUMN row_number;",temporaryTable));
+    queries.add(
+        String.format("COPY %s FROM '%s' DELIMITER ',' CSV HEADER;", temporaryTable, filePath));
+    queries.add(
+        String.format("INSERT INTO %s SELECT * FROM %s;", destinationTable, temporaryTable));
+    queries.add(String.format("DROP TABLE %s;", temporaryTable));
+    queries.add(
+        String.format("ALTER TABLE \"%s\" ADD COLUMN row_number SERIAL;", destinationTable));
+    return queries;
+  }
+  /**
+   * Generate the query for point in time correctness join of data for a single feature set to the
+   * entity dataset.
+   *
+   * @param featureSetInfo Information about the feature set necessary for the query templating
+   * @param leftTableName entity dataset name
+   * @param minTimestamp earliest allowed timestamp for the historical data in feast
+   * @param maxTimestamp latest allowed timestamp for the historical data in feast
+   * @return point in time correctness join BQ SQL query
+   */
+  protected String createFeatureSetPointInTimeQuery(
+      FeatureSetQueryInfo featureSetInfo,
+      String leftTableName,
+      String minTimestamp,
+      String maxTimestamp)
+      throws IOException {
+    PebbleTemplate template;
+    template = engine.getTemplate(FEATURESET_TEMPLATE_NAME_POSTGRES);
+
+    Map<String, Object> context = new HashMap<>();
+    context.put("featureSet", featureSetInfo);
+
+    // TODO: Subtract max age to min timestamp
+    context.put("minTimestamp", minTimestamp);
```

Comment:
it seems like `minTimestamp` is never used in the FEATURESET_TEMPLATE? Only `maxTimestamp` is using in the SQL FEATURESET_TEMPLATE  to check `WHERE event_timestamp <= '{{maxTimestamp}}'`

Reply:
Also, I think I need some clarification on `TODO: Subtract max age to min timestamp`.